### ISSUE TITLE
TestFlags are not saved in ~/.cabal/config

### ISSUE THREAD

**Issue by melrief** [haskell/cabal#1941]
[TestFlags](https://github.com/haskell/cabal/blob/master/Cabal/Distribution/Simple/Setup.hs#L1569) are not saved in `~/.cabal/config` and must be passed by command line.

Related to this [StackOverflow question](http://stackoverflow.com/a/24211177/1095506).

**Comment by Abastro** [comment]
Hello, I would like this feature implemented. What are the blockers for this feature?

**Comment by Abastro** [comment]
@ulysses4ever Oh, is this one hard to implement?
It seems like the linked `Cabal/Distribution/Simple/Setup.hs` no longer exists, so I guess it would be done a bit differently?

**Comment by Mikolaj** [comment]
Probably not that hard and, in any case, worth a PR, with a test, with some experiments, and then we'd see how to goes.

I'd guess it's this file?

```
~/r/cabal$ find . -name "*Setup.hs"
./cabal-dev-scripts/Setup.hs
./Cabal/src/Distribution/Simple/Setup.hs
...
```

**Comment by Mikolaj** [comment]
BTW, things like `--show-details=always` from the original request now go into `cabal.project` and work fine. But perhaps not everything does; worth checking.

**Comment by Abastro** [comment]
Oh. The behavior seems not documented, so I assumed it does not exist.. Thanks a lot!

**Comment by ulysses4ever** [comment]
I'm guessing that if it works from a project file, than it should work from the config file too? (Although, this is not at simple I'm general #6319)

**Comment by Abastro** [comment]
Can I ask where I could find list of options supported by `cabal.project.local` in the documentation?

**Comment by ulysses4ever** [comment]
@Abastro 
> Can I ask where I could find list of options supported by cabal.project.local in the documentation?

In the manual: [7. cabal.project Reference](https://cabal.readthedocs.io/en/stable/cabal-project.html).

By the way, I tried to put it in `~/.cabal/config` instead of `cabal.project.local` -- and it didn't work. This is surprising to me, as many options work from either of the files.

**Comment by Abastro** [comment]
I do not see `test-show-details` in the list, while it is a valid option in cabal.project.local. Maybe documentation should be updated to reflect the options?

**Comment by ulysses4ever** [comment]
Correct: the manual is in no way complete. (What's even funnier is that `--test-show-details` is not documented at all! #8619) The rule is thumb is that whatever goes on command line can go to cabal.project(.local) in some form.
